embassy-rp/src/clocks.rs

```diff
@@ -2,10 +2,139 @@ use pac::clocks::vals::*;
 
 use crate::{pac, reset};
 
-const XOSC_MHZ: u32 = 12;
+// TODO fix terrible use of global here
+static mut XIN_HZ: u32 = 0;
+
+pub struct ClockConfig {
+    rosc_config: Option<RoscConfig>,
+    xosc_config: Option<XoscConfig>,
+    ref_clk_config: (RefClkSrc, u8),
```

Comment:
i'd use structs here instead of tuples. With tuples you can't name the fields, so it ends up being not very self-documenting. There's no way to know the `u8` is the divider just from reading the types.

Reply:
I was going back and forth on this. Partially because I felt the huge number of structs was a bit too verbose. But, I don't have a super strong preference so I switched it.